### ISSUE TITLE
Conan integration

### ISSUE THREAD

**Issue by AndreyMlashkin** [rui314/mold#570]
FYI: I am currently working on including mold as a build tool to conan
github.com/conan-io/conan-center-index/pull/11536

I have tried to build some of the most popular libraries with mold + clang and it worked, but I am getting a failure with glib.
If you want to reproduce error:
- install conan with 'pip install conan'
- create a build profile in ~/.conan/profiles/default and fill it similar to mine:
```[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.cppstd=17
compiler.libcxx=libstdc++
compiler.version=14
os=Linux
os_build=Linux
[options]
*:shared=False
[build_requires]
[env]
CC=/usr/bin/clang
CFLAGS=-fuse-ld=/home/andrei/sources/mold/mold
CXX=/usr/bin/clang++
CXXFLAGS=-fuse-ld=/home/andrei/sources/mold/mold
```
- try to compile glib with 'conan install  glib/2.73.0@ --build glib'

it fails with error:
```
Checking if "GCC size_t typedef is long" compiles: NO 
source_subfolder/meson.build:1560:2: ERROR: Problem encountered: Could not determine size of size_t.
```
Logs are attached
[glib_mold_fail.txt](https:[meson-log.txt](https://github.com/rui314/mold/files/9041160/meson-log.txt)github.com/rui314/mold/files/9041149/glib_mold_fail.txt)

**Comment by AndreyMlashkin** [comment]
[glib_mold_fail.txt](https://github.com/rui314/mold/files/9041164/glib_mold_fail.txt)

**Comment by eli-schwartz** [comment]
> A full log can be found at /home/andrei/.conan/data/glib/2.73.0/_/_/build/7dcbe832e6591285c06820413549b7979b27e927/build_subfolder/meson-logs/meson-log.txt

This file is relevant to figure out what Meson actually ran that produced this error.

**Comment by rui314** [comment]
I got the following error when I run `conan install glib/2.73.0@ --build glib`. Is something missing in your config file? 

```
root@3657bfea52d5:~# conan install glib/2.73.0@ --build glib
Configuration:
[settings]
[options]
*:shared=False
[build_requires]
[env]
CC=/usr/bin/clang
CFLAGS=-fuse-ld=/root/mold/mold
CXX=/usr/bin/clang++
CXXFLAGS=-fuse-ld=/root/mold/mold
ERROR: glib/2.73.0: 'settings.arch' value not defined
```

**Comment by AndreyMlashkin** [comment]
@rui314 sorry, in my initial comment missing

**[settings]**
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.cppstd=17
compiler.libcxx=libstdc++
compiler.version=14
os=Linux
os_build=Linux
[options]
*:shared=False
[build_requires]
[env]
CC=/usr/bin/clang
CFLAGS=-fuse-ld=/home/andrei/sources/mold/mold
CXX=/usr/bin/clang++
CXXFLAGS=-fuse-ld=/home/andrei/sources/mold/mold

**Comment by AndreyMlashkin** [comment]
@eli-schwartz here is the log:
[meson-log.txt](https://github.com/rui314/mold/files/9044150/meson-log.txt)

**Comment by rui314** [comment]
@AndreyMlashkin  Now I got the following new error.

```
root@3657bfea52d5:~# conan install glib/2.73.0@ --build glib
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.cppstd=17
compiler.libcxx=libstdc++
compiler.version=14
os=Linux
os_build=Linux
[options]
*:shared=False
[build_requires]
[env]
CC=/usr/bin/clang
CFLAGS=-fuse-ld=/root/mold/mold
CXX=/usr/bin/clang++
CXXFLAGS=-fuse-ld=/root/mold/mold
glib/2.73.0: Forced build from source
Installing package: glib/2.73.0
Requirements
    bzip2/1.0.8 from 'conancenter' - Cache
    glib/2.73.0 from 'conancenter' - Cache
    libelf/0.8.13 from 'conancenter' - Cache
    libffi/3.4.2 from 'conancenter' - Cache
    libmount/2.36.2 from 'conancenter' - Cache
    libselinux/3.3 from 'conancenter' - Cache
    pcre/8.45 from 'conancenter' - Cache
    pcre2/10.37 from 'conancenter' - Cache
    zlib/1.2.12 from 'conancenter' - Cache
Packages
    bzip2/1.0.8:83dda2038feb1d4fd2af9d22e7ded1ece671c569 - Missing
    glib/2.73.0:7dcbe832e6591285c06820413549b7979b27e927 - Build
    libelf/0.8.13:e2776531edc78fdd251990dfaea66391e6d3a3c1 - Missing
    libffi/3.4.2:e2776531edc78fdd251990dfaea66391e6d3a3c1 - Missing
    libmount/2.36.2:e2776531edc78fdd251990dfaea66391e6d3a3c1 - Missing
    libselinux/3.3:47e9e2b020a57d7af7586c5810bfd35d48d26658 - Missing
    pcre/8.45:fd5c56e2aed3f5c286a1070dd88aa1e00e3ddbf7 - Missing
    pcre2/10.37:962beb8ac6d41edbd528a1b29c7a4ba713c94ccd - Missing
    zlib/1.2.12:e2776531edc78fdd251990dfaea66391e6d3a3c1 - Missing
Build requirements
    meson/0.61.2 from 'conancenter' - Cache
    ninja/1.10.2 from 'conancenter' - Cache
    pkgconf/1.7.4 from 'conancenter' - Cache
Build requirements packages
    meson/0.61.2:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    ninja/1.10.2:cc9f9b2b9f92fc82888026414560bf79d8c3a052 - Download
    pkgconf/1.7.4:cc9f9b2b9f92fc82888026414560bf79d8c3a052 - Download

Installing (downloading, building) binaries...
ERROR: Missing binary: bzip2/1.0.8:83dda2038feb1d4fd2af9d22e7ded1ece671c569
ERROR: Missing binary: libelf/0.8.13:e2776531edc78fdd251990dfaea66391e6d3a3c1
ERROR: Missing binary: libffi/3.4.2:e2776531edc78fdd251990dfaea66391e6d3a3c1
ERROR: Missing binary: libmount/2.36.2:e2776531edc78fdd251990dfaea66391e6d3a3c1
ERROR: Missing binary: libselinux/3.3:47e9e2b020a57d7af7586c5810bfd35d48d26658
ERROR: Missing binary: pcre/8.45:fd5c56e2aed3f5c286a1070dd88aa1e00e3ddbf7
ERROR: Missing binary: pcre2/10.37:962beb8ac6d41edbd528a1b29c7a4ba713c94ccd
ERROR: Missing binary: zlib/1.2.12:e2776531edc78fdd251990dfaea66391e6d3a3c1

bzip2/1.0.8: WARN: Can't find a 'bzip2/1.0.8' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Debug, compiler=clang, compiler.version=14, os=Linux
- Options: build_executable=True, fPIC=True, shared=False
- Dependencies:
- Requirements:
- Package ID: 83dda2038feb1d4fd2af9d22e7ded1ece671c569

ERROR: Missing prebuilt package for 'bzip2/1.0.8', 'libelf/0.8.13', 'libffi/3.4.2', 'libmount/2.36.2', 'libselinux/3.3', 'pcre/8.45', 'pcre2/10.37', 'zlib/1.2.12'
Use 'conan search bzip2/1.0.8 --table=table.html -r=remote' and open the table.html file to see available packages
Or try to build locally from sources with '--build=missing'

More Info at 'https://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package'
```

**Comment by eli-schwartz** [comment]
@rui314 as I said above, *all* information needed to determine the cause of the issue is in the log file I told @AndreyMlashkin to look at.

Since the log file has been uploaded, you can look at that log file too and see the problem as well. It's right there, on the last few lines of the log, and shows that this is absolutely positively not a mold bug.

(Mold is a linker, so it was never going to have any effect on checking if something compiles as `clang -c foo.c -o foo.obj`.)

**Comment by AndreyMlashkin** [comment]
@rui314 to build missing dependencies with Conan, you should add '--build missing' to the command

**Comment by eli-schwartz** [comment]
Consider the error message. The error message is saying that ***clang*** does not allow you to specify a linker in CFLAGS. It doesn't matter what linker -- mold, bfd, gold, lld, regardless of which you choose, they will all error out.

Clang simply objects that if you put it in CFLAGS (or CXXFLAGS) it will be "unused" when compiling.

This is because choosing a linker is a *link* option, not a *compile* option.

Try putting it into `$LDFLAGS` instead.

**Comment by AndreyMlashkin** [comment]
changing LDFLAGS helps, thank you!

**Comment by AndreyMlashkin** [comment]
Btw, it would be very helpful, if you add CMakeLists to the project.

**Comment by Alcaro** [comment]
It will be closed, yes, but not for that reason. It'd be closed as duplicate of #571.

**Comment by AndreyMlashkin** [comment]
It seems to be some general problem with macos + arm arch. I have disabled it for now.
mold will be soon available as a build tool in conan. My PR is in a review process.

Looking forward to a new version, which will get windows support
